src/Feldspar/Compiler/Backend/C/CodeGeneration.hs

```diff
@@ -75,14 +75,12 @@ instance CodeGen (Entity ())
       | otherwise = start <> semi
         where
          start
-           | loopBody  = text ("LOOP_BODY_" ++ show (length inParams + length outs - 1))
+           | loopBody  = text ("LOOP_BODY_" ++ show (length inParams - 1))
                           <> parens (sep $ punctuate comma loopH)
-           | otherwise = rtype <+> text procName <> parens (pvars env $ inParams ++ outs)
-         (outs, rtype)
-           | Left params <- outParams = (params, text "void")
-           | Right param <- outParams = ([], cgen env (typeof param))
+           | otherwise = rtype <+> text procName <> parens (pvars env inParams)
+         rtype = cgen env procType
          loopH = [text procName, text "LARGE_BODY"] ++
-                  concatMap (\v ->  [cgen env (typeof v), cgen env v]) (inParams ++ outs)
+                  concatMap (\v ->  [cgen env (typeof v), cgen env v]) inParams
```

Comment:
I think this line cares about the variable name.

Reply:
Yes, but the intention is that the outParams variables will be included. The error is rather in the commit message.